### DISCUSSION TITLE
SSE visualization ideas

### DISCUSSION THREAD

**Discussion by OLaprevote** [BradyAJohnston/MolecularNodes#202] [Ideas]
When I realized yesterday that all the SSE detection was on main I had a go at it. Got a few ideas/opinions that may be worth sharing:

I only had a short glance at how the arrow-head of beta-sheets are done and had the impression that it basically inserts a triangle at the end. Would it be possible instead to do it by suddenly increasing the width of the mesh primitive at the end of each beta-sheet and then linearly decrease it until reaching the loop ? I think it would kinda keep following the curve and would look less standing out/rigid. Not sure how feasible it is though.

I may be wrong but I think it would be nice to make the mesh primitive for secondary structures more apparent. Right now there's some digging to do before arriving there. It could allow to get quite creative for the representation (I'm thinking of visuals such as [here](https://twitter.com/SciCommStudios/status/1245345852782190593), [here](https://twitter.com/SciCommStudios/status/1215658689690062852/photo/1) or [there](https://twitter.com/luminous_lab/status/1253746784293269505), but hey one could start doing churros-shaped loops as well)

**Comment by BradyAJohnston** [comment]
The changing of the width to get the arrow head can certainly be done. It's how I originally approached it, but it didn't look great because of the approach that I was taking with the rest of the strand wasn't sharp / square enough so I went with the physical arrow head. It can certainly be done and thinking about it I know how I would do it a bit nicer that should work better. I certainly like the look of the examples that you have given.

I would like to modularise the mesh creation a bit more, so that in theory someone could apply the arrow to any arbitrary curve. Will have a play around with making it a bit more modular and reuseable.
